mobile_app/openSleep/Recordings.swift

```diff
@@ -180,38 +254,196 @@ class RecordingsManager : NSObject, AVAudioRecorderDelegate, AVAudioPlayerDelega
+  }
+  
+  /*
+    Resets silencedetection-related parameters
+ */
+  func resetSilenceDetection() {
+    print("resetting silence detection")
+    silenceTime = 0.0
+    recordingTimeElapsed = 0.0
+    silenceDetectionTimer.invalidate()
+  }
+  
+  /*
+   Starts an audio recording to use as a reference for silence. Recording continues for calibrationTime variable.
+   Average power is calculated at the end of the recording session, and is stored in calibrationSilenceThresh, used in silence detection
+ */
+  func calibrateSilenceThreshold() {
+    let fileManager = FileManager.default
+    let urls = fileManager.urls(for: .documentDirectory, in: .userDomainMask)
+    let documentDirectory = urls[0] as NSURL
+    let url = documentDirectory.appendingPathComponent("calibrateSilentThreshold.m4a")!
+    
+    do {
+    audioRecorder = try AVAudioRecorder(url: url as URL,
+                                        settings: audioRecorderSettings)
+    audioRecorder.delegate = self
+    audioRecorder.isMeteringEnabled = true
+    audioRecorder.prepareToRecord()
+      
+    } catch {
+      audioRecorder.stop()
+    }
+    audioRecorder.record()
+    
+    var runningSum:Float = 0.0
+    
+    calibrateTimer = Timer.scheduledTimer(withTimeInterval: Double(self.calibrationTimeStep), repeats: true, block: { (timer: Timer) in
+      if(self.elapsedCalTime > self.calibrationTime) {
+        timer.invalidate()
+        self.audioRecorder.stop()
+        self.calibrationSilenceThresh = Float(runningSum / (self.calibrationTime / self.calibrationTimeStep))
+        print("CALIBRATION SILENCE THRESH IS", self.calibrationSilenceThresh)
+        self.elapsedCalTime = 0.0
+      } else {
+        self.elapsedCalTime += 0.1
+        self.audioRecorder.updateMeters()
+        runningSum += self.audioRecorder.averagePower(forChannel: 0)
+      }
+    })
+    
+  }
+  
+  /*
+    Resets audio parameters and related timers
+ */
+  func reset() {
+    silenceTime = 0.0
+    recordingTimeElapsed = 0.0
+    elapsedCalTime = 0.0
+    calibrateTimer.invalidate()
+    silenceDetectionTimer.invalidate()
     audioRecorder.stop()
   }
   
   func startPlaying(mode: Int, onFinish: (() -> ())? = nil) {
-    self.audioPlayer = try! AVAudioPlayer(contentsOf: audioURLs[mode]!)
-    self.audioPlayer.prepareToPlay()
-    self.audioPlayer.delegate = self
-    //    self.audioPlayer.currentTime = max(0 as TimeInterval, self.audioPlayer.duration - audioPlaybackOffset)
-    self.audioPlayer.play()
+    if let url = audioURLs[mode] {
+      self.audioPlayer = try! AVAudioPlayer(contentsOf: url)
+      self.audioPlayer.prepareToPlay()
+      self.audioPlayer.delegate = self
+      //    self.audioPlayer.currentTime = max(0 as TimeInterval, self.audioPlayer.duration - audioPlaybackOffset)
+      self.audioPlayer.play()
+    } else {
+      return
+    }
+  }
+  
+  // To Eyal: again I left in both startPlaying and startPlayingMulti, should i leave as is or combine them
```

Comment:
Again, I think this is OK. Maybe just choose a different naming convention - like in the recording function, where you have recordDream and recordMulti (although maybe change recordMulti to recordMessages? recordInterruptions? something of the sort)

So also rename the function and variables here to fit with the above, and it will be clear why there are two function

Reply:
Yea the naming could be improved

Reply:
Actually, I think I want to keep Multi as the suffix, in the case we decide we want multiple prompt messages as well.